packages/astro/src/core/util.ts

```diff
@@ -18,6 +18,9 @@ export function isURL(value: unknown): value is URL {
 }
 /** Check if a file is a markdown file based on its extension */
 export function isMarkdownFile(fileId: string, option?: { suffix?: string }): boolean {
+	if (fileId.endsWith('?raw')) {
+		return false;
+	}
```

Comment:
Does `?raw` have to be at the end of the query? What if it's in the middle?

Reply:
Do we need to handle other params? We have https://github.com/withastro/astro/blob/0a0b1978a7ea9902174df96852e6a676023cd128/packages/astro/src/vite-plugin-utils/index.ts#L59

Reply:
I covered it a bit in the PR description. In theory I _think_ it could go elsewhere in the query, but I'm not aware of a scenario where it could legitimately combined with any other. It conflicts with all of the other suffixes that I know of (`worker`, `shared-worker`, `url`). There may be ones I'm missing, and of course parsing it fully as a query string would be more robust, but the question is whether it's worth doing that for every import when it's such an edgy of edge cases.

Reply:
I've switched it to use `hasSpecialQueries`